src/types.ts

```diff
@@ -184,10 +190,26 @@ export function packedType(type: FieldDescriptorProto_Type): number | undefined
   }
 }
 
+export function getFieldOptionsJsType(
+  field: FieldDescriptorProto,
+  options: Options,
+): FieldDescriptorProto_Type | undefined {
+  if (!options.useJsTypeOverride || field.options?.jstype === undefined) {
+    return;
+  }
+
+  switch (field.options.jstype) {
+    case FieldOptions_JSType.JS_STRING:
+      return FieldDescriptorProto_Type.TYPE_STRING;
+    case FieldOptions_JSType.JS_NUMBER:
+      return FieldDescriptorProto_Type.TYPE_INT64;
```

Comment:
Do we need to handle the JS_NORMAL value here?

https://googleapis.dev/nodejs/nodejs-functions/latest/google.protobuf.FieldOptions.html

If not, maybe put an in-source comment about why not? Like maybe if `JS_NORMAL` is used, the `jstype` ends up being `undefined` because `JS_NORMAL` is an 0th value enum (just guessing!)...

Reply:
From my understanding, the `JS_NORMAL` case would be equivalent to not having the `JSType` set on the field at all, so we want the behaviour to match the same case as if it were undefined. That's why I made it fallthrough and default to not returning an explicit value.
I'll add a quick comment explaining this 👍
I could also maybe move it into the guard clause above